### ISSUE TITLE
401 Unauthorized - conflict with other dependencies

### ISSUE THREAD

**Issue by yeryomenkom** [DanielaSfregola/twitter4s#125]
I am trying to run simple snippet of code:

object TwitterSupplier extends App {

  val client = TwitterStreamingClient()

  def printTweetText: PartialFunction[StreamingMessage, Unit] = {
    case tweet: Tweet => println(tweet.text)
  }

  client.sampleStatuses(stall_warnings = true)(printTweetText)

}

Tokens are in application.conf
Always get the same result: 
ERROR twitter4s - Stream could not be opened: com.danielasfregola.twitter4s.exceptions.TwitterException: [401 Unauthorized] <html>\n<head>\n<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>\n<title>Error 401 Unauthorized</title>
</head>
<body>
<h2>HTTP ERROR: 401</h2>
<p>Problem accessing '/1.1/statuses/sample.json?stall_warnings=true'. Reason:
<pre>    Unauthorized</pre>
</body>
</html>
 (-1)

I tried to regenerate tokens and repeat process, but result still the same.

**Comment by DanielaSfregola** [comment]
Hi @yeryomenkom,
just tested the code you posted and it works just fine for me.

It must be a problem associated with your tokens.

A few tips that could help you find out where the problem is:
- try to use any call from `TwitterRestClient`. If even the call using the `TwitterRestClient` is failing with a 401, there is a good chance that your tokens are not quite right.
- check that your configuration in the `application.conf` in not overridden by any env variable -- the library will give the precedence to env variables over configuration files
- double check that you haven't swapped consumer with access token of the key with the secret: you need to have 4 total token values (i.e.: consumer_key, consumer_secret, access_key, access_secret).

Cheers,
D.

**Comment by yeryomenkom** [comment]
Thank your for reply)
I've tried use `TwitterRestClient` and get the same error. So this is something wrong with my keys.
access_key and access_secret are Access Token and Access Token Secret from https://apps.twitter.com page of my app?
Or maybe I should specify Callback URL while creating twitter app?
I checked my keys few times, regenerate they, tried again and again and nothing work :(

**Comment by yeryomenkom** [comment]
Is there libraries, which could conflict with twitter4s?
I've removed all other dependencies from my project and now everything works fine.
Maybe some of them could: akkaClusterTools, akkaHttp, akkaSse?

**Comment by DanielaSfregola** [comment]
Hi @yeryomenkom,
glad this is working now.

I do not believe other dependencies are able to conflict with twitter4s.

the 401 message means that you were able to connect to twitter, it just means they didn't like the authentication provided.

What happens if you print out the tokens are initializing the client? Are they different with or without these dependencies?

For example you could do: 
```
val client = TwitterStreamingClient()
println(client.consumerToken)
println(client.accessToken)
```

Note: PLEASE, DO NOT SHARE THE TOKENS HERE! 💃

**Comment by yeryomenkom** [comment]
Problem occurs when adding ` "com.typesafe.akka" %% "akka-http" % "10.0.3"` to dependencies list.
I've checked tokens. There are the same.

**Comment by EhsanYousefi** [comment]
Same issue, I've double checked everything, I'm gonna try another client, maybe twitter is the problem.

**Comment by EhsanYousefi** [comment]
Twitter4j works fine with the same credentials, so nothing is wrong with Twitter.

**Comment by DanielaSfregola** [comment]
Hi @EhsanYousefi, as @yeryomenkom said it is a problem with overriding dependencies.

I'll have a look and fix it asap. In the meantime, contributions are more than welcome.

**Comment by bbarker** [comment]
Just confirming we have the same problem with using akka-http 10.0.3. Had been baffled by this for a while and was just about to post an issue since the problem didn't occur with a minimum working example and with the same credentials (as reported by others)!

**Comment by bbarker** [comment]
Is there a good way to get an http dump in twitter4s? Might be helpful for comparison with the minimal working example.

**Comment by DanielaSfregola** [comment]
Hi guys,
thanks again for raising this issue!

Found and fix the bug (see #127 if you are curious). This indeed a pretty annoying problem, so I have decided to release it right away.

Upgrading to `twitter4s 5.1` should fix the issue, so please do so and let me know if the issue still persists.

**Comment by DanielaSfregola** [comment]
Hi @bbarker, 
if you enable the debugger you should be able to see all the details of the requests made by twitter4s. 

Cheers,
D.
